public/app/features/apiserver/client.ts

```diff
@@ -34,6 +38,43 @@ export class ScopedResourceClient<T = object, S = object, K = string> implements
     return getBackendSrv().get<Resource<T, S, K>>(`${this.url}/${name}`);
   }
 
+  public watch(opts?: WatchOptions): Observable<ResourceEvent<T, S, K>> {
+    const finalOpts = opts || {};
+    finalOpts.labelSelector = this.parseListOptionsSelector(finalOpts?.labelSelector);
+    finalOpts.fieldSelector = this.parseListOptionsSelector(finalOpts?.fieldSelector);
+    return getBackendSrv()
+      .chunked({
+        url: finalOpts.name ? `${this.url}/${finalOpts.name}` : this.url,
+        params: finalOpts,
+      })
+      .pipe(
+        switchMap((response) => {
+          const events: Array<ResourceEvent<T, S, K>> = [];
+          if (response.ok && response.data) {
+            const decoder = new TextDecoder();
+            decoder
+              .decode(response.data)
+              .split('\n')
```

Comment:
Do you think we need to handle the case where chunks could have incomplete JSON? E.g., it gets split in the middle or something like that.

Reply:
The supported javascript client parses each line... see: https://github.com/kubernetes-client/javascript/blob/1.0.0/src/watch.ts#L65

I don't know if their 'lines' support is more complicated than looking for `\n` 🤔